eZ/Bundle/EzPublishIOBundle/Migration/FileListerRegistry/IdentifierBased.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+/**
+ * File containing the IdentifierBased class.
+ *
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace eZ\Bundle\EzPublishIOBundle\Migration\FileListerRegistry;
+
+use eZ\Bundle\EzPublishIOBundle\Migration\FileListerRegistry;
+use eZ\Publish\Core\Base\Exceptions\NotFoundException;
+
+/**
+ * A registry of FileListerInterfaces that uses an identifier string to identify the lister.
+ */
+final class IdentifierBased implements FileListerRegistry
```

Comment:
The naming could be improved here, in regards to interface/implementation: the interface _is_ already about identifiers, since it has `getItem($identifier)` and `getIdentifiers()` methods.

It could be a ConfigurableRegistry (as you configure it by means of its constructor arguments), or maybe ArrayBasedRegistry, as it uses an array internally.

What do you think ?